update maturity dates for require_generic_test_arguments_property flag (#7758)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/behavior-changes.md

Reviewer: Should we also edit the default value in line 67?

dbt supports parsing key-value arguments that are inputs to generic tests when specified under the `arguments` property. In the past, dbt didn't support a way to clearly disambiguate between properties that were inputs to generic tests and framework configurations, and only accepted arguments as top-level properties.

> By default, the `require_generic_test_arguments_property` flag is set to `True` in "Latest", and `False` in versions of dbt Core less than 1.11.0.

Reviewer: ```suggestion
By default, the `require_generic_test_arguments_property` flag is set to `True` in "Latest". In dbt Core versions prior to 1.11.0, the default value is `False` .
```

> Usage of the `arguments` property in test definitions is optional in either case.

Reviewer: ```suggestion
Using the `arguments` property in test definitions is optional in either case.
```

> If you do use `arguments` while the flag is `False`, dbt will recognize it but raise the `ArgumentsPropertyInGenericTestDeprecation` warning to let you know that the flag will eventually default to `True` across all releases and be parsed as keyword arguments to the test.

Reviewer: ```suggestion
If you do use `arguments` while the flag is `False`, dbt will recognize it but raise the `ArgumentsPropertyInGenericTestDeprecation` warning. This warning lets you know that the flag will eventually default to `True` across all releases and will be parsed as keyword arguments to the data test.
```
